### PR TITLE
Allow the user to choose the start address of the delivery.

### PR DESCRIPTION
**Pull request by ak47na**
The start address is either his current location retrieved using Geolocation, or an adress written by the user when making the delivery slot request.

Note: This branch will be merged after branch singleDeliveryRequest.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/NewRequestServlet.java [hunk 1]
```diff
@@ -89,10 +90,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
           request.getParameter("end-time"),
           request.getParameter("max-stops"),
           userId);
+
+      if (request.getParameter("latitude") == null || request.getParameter("latitude") == null) {
```
Comment (ghost) [inline src/main/java/com/google/sps/servlets/NewRequestServlet.java]: Can you move these magic strings to a constant in this class?  latitude, longitude, start-address etc

File: src/main/java/com/google/sps/servlets/NewRequestServlet.java [hunk 2]
```diff
@@ -89,10 +90,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+      // the user has set the address himself
+      slotManager.setSlotStartingPoint(request.getParameter("start-address"));
```
Comment (ghost) [inline src/main/java/com/google/sps/servlets/NewRequestServlet.java]: Nit: indentation seems off in the diff view

File: src/main/webapp/common.js [hunk 3]
```diff
@@ -95,9 +95,14 @@ function updateAddressFromGeolocation(checkboxId, addressId) {
         }).catch((error) => geolocationError(error));
     } else {
       // clear the address box if location is not supported 
-      addressBox.value = ""; 
+      document.getElementById(addressId).value = "";
```
Comment (ghost) [inline src/main/webapp/common.js]: Nit: please use single quotes for string literals - https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes

```suggestion
      document.getElementById(addressId).value = ''; 
```

File: src/main/webapp/common.js [hunk 4]
```diff
@@ -95,9 +95,14 @@ function updateAddressFromGeolocation(checkboxId, addressId) {
         }).catch((error) => geolocationError(error));
     } else {
       // clear the address box if location is not supported 
-      addressBox.value = ""; 
+      document.getElementById(addressId).value = ""; 
       alert("Geolocation is not supported by this browser!");
     }
+  } else {
+    // clear the address box to allow the user to set it.
+    document.getElementById(addressId).value = "";
```
Comment (ghost) [inline src/main/webapp/common.js]:
```suggestion
    document.getElementById(addressId).value = ''; 
```

**Review by ghost — APPROVED** [review]

**Review by anamariamilcu — APPROVED** [review]

File: src/main/webapp/common.js (hunk 3 above)
Comment (ak47na) [reply src/main/webapp/common.js]: I changed them.

File: src/main/java/com/google/sps/servlets/NewRequestServlet.java (hunk 2 above)
Comment (ak47na) [reply src/main/java/com/google/sps/servlets/NewRequestServlet.java]: Done.

File: src/main/java/com/google/sps/servlets/NewRequestServlet.java (hunk 1 above)
Comment (ak47na) [reply src/main/java/com/google/sps/servlets/NewRequestServlet.java]: I removed them and added a DeliverySlotParameters enum.

**Pull request closed without merging**